catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/DeleteOperations.java

```diff
@@ -311,6 +317,51 @@ public DeleteResponse delete(DeleteRequest deleteRequest)
     //
     // Private helper methods
     //
+    private DeleteRequest rewriteQuery(DeleteRequest deleteRequest)
```

Comment:
This needs a better name.  It should be clear that this method is verifying that the attributes are unique but it will rewrite the query if they are not unique due to history matches.